aws/config.go

```diff
@@ -38,31 +38,52 @@ type Config struct {
 	// will prevent the SDK from modifying the HTTP client.
 	HTTPClient HTTPClient
 
-	// An endpoint resolver that can be used to provide or override an endpoint for the given
-	// service and region.
+	// An endpoint resolver that can be used to provide or override an endpoint
+	// for the given service and region.
 	//
-	// See the `aws.EndpointResolver` documentation for additional usage information.
+	// See the `aws.EndpointResolver` documentation for additional usage
+	// information.
 	//
 	// Deprecated: See Config.EndpointResolverWithOptions
 	EndpointResolver EndpointResolver
 
-	// An endpoint resolver that can be used to provide or override an endpoint for the given
-	// service and region.
+	// An endpoint resolver that can be used to provide or override an endpoint
+	// for the given service and region.
 	//
-	// When EndpointResolverWithOptions is specified, it will be used by a service client rather than using
-	// EndpointResolver if also specified.
+	// When EndpointResolverWithOptions is specified, it will be used by a
+	// service client rather than using EndpointResolver if also specified.
 	//
-	// See the `aws.EndpointResolverWithOptions` documentation for additional usage information.
+	// See the `aws.EndpointResolverWithOptions` documentation for additional
+	// usage information.
 	EndpointResolverWithOptions EndpointResolverWithOptions
 
-	// Retryer is a function that provides a Retryer implementation. A Retryer guides how HTTP requests should be
-	// retried in case of recoverable failures. When nil the API client will use a default
-	// retryer.
+	// RetryMaxAttempts specifies the maximum number attempts an API client
+	// will call an operation that fails with a retryable error.
 	//
-	// In general, the provider function should return a new instance of a Retryer if you are attempting
-	// to provide a consistent Retryer configuration across all clients. This will ensure that each client will be
-	// provided a new instance of the Retryer implementation, and will avoid issues such as sharing the same retry token
-	// bucket across services.
+	// API Clients will only use this value to construct a retryer if the
+	// Config.Retryer member is not nil. This value will be ignored if
+	// Retryer is not nil.
+	RetryMaxAttempts int
+
+	// RetryMode specifies the retry model the API client will be created with.
+	//
+	// API Clients will only use this value to construct a retryer if the
+	// Config.Retryer member is not nil. This value will be ignored if
+	// Retryer is not nil.
+	RetryMode RetryMode
```

Comment:
Thoughts on whether these should be grouped into a `RetryOptions` structure or should this remain stand-alone? Also it looks like this doesn't contain the wiring up of using the retry mode specified by defaults configuration. We should make sure we add the plumbing to wire-up the retryer with the retry mode specified from the SDK's configured DefaultsMode when this `RetryMode` on `aws.Config` is not set.

Reply:
Yeah keep these top level since `Retryer` was already here. If `Retryer` was within some `RetryOptions` would of used that.

Will look at what overlap there is for Default configurations with retry, and get those wired up. I think this should be at the API client resolve level, with any of these specific parameters overriding any DefaultConfig options.